PerfTools/AllocMonitor/plugins/ModuleEventAllocMonitor.cc

```diff
@@ -0,0 +1,445 @@
+        auto index = mod_id;
+        if (not moduleIDs_.empty()) {
+          auto it = std::lower_bound(moduleIDs_.begin(), moduleIDs_.end(), mod_id);
+          index = it - moduleIDs_.begin();
+        }
+        {
+          auto const& alloc = streamModuleAllocs_[nModules_ * iStream.streamID().value() + index];
+          assert(alloc.size() == alloc.allocationSizes().size());
+        }
+        streamModuleAllocs_[nModules_ * iStream.streamID().value() + index] = info->allocMap_;
+        {
+          auto const& alloc = streamModuleAllocs_[nModules_ * iStream.streamID().value() + index];
+          assert(alloc.size() == alloc.allocationSizes().size());
+        }
+        ++streamSync_[iStream.streamID().value()];
+        streamModuleInAcquire_[nModules_ * iStream.streamID().value() + index].store(false);
+      }
+    });
+    /*
+    iRegistry.watchPreModuleEventDelayedGet(
+                                            StreamEDModuleState<Step::preModuleEventDelayedGet>(logFile, beginTime, iFilter));
+    iRegistry.watchPostModuleEventDelayedGet(
+                                             StreamEDModuleState<Step::postModuleEventDelayedGet>(logFile, beginTime, iFilter));
+    iRegistry.watchPreEventReadFromSource(
+                                          StreamEDModuleState<Step::preEventReadFromSource>(logFile, beginTime, iFilter));
+    iRegistry.watchPostEventReadFromSource(
+                                           StreamEDModuleState<Step::postEventReadFromSource>(logFile, beginTime, iFilter));
+    */
+    iAR.watchPreClearEvent([this](auto const& iStream) { filter_.startOnThread(); });
+    iAR.watchPostClearEvent([this, file](auto const& iStream) {
+      auto info = filter_.stopOnThread();
+      if (info) {
+        //auto v = std::accumulate(info->allocMap_.allocationSizes().begin(),
+        //                         info->allocMap_.allocationSizes().end(),
+        //                        0);
+        //edm::LogSystem("MEA")<< "temp mem size "<<info->totalMatchedDeallocSize_ <<" # "<<info->numMatchedDeallocs_ <<"\nold mem size "<<info->totalUnmatchedDealloc_<<" # "<<info->numUnmatchedDeallocs_ <<
+        //  "\nnew mem size "<< v <<" # "<<info->allocMap_.allocationSizes().size();
+        streamSync_[iStream.streamID().value()].load();
+        auto itBegin = streamModuleAllocs_.begin() + nModules_ * iStream.streamID().value();
+        auto const itEnd = itBegin + nModules_;
+
+        {
+          std::vector<std::size_t> moduleDeallocSize(nModules_);
+          std::vector<unsigned int> moduleDeallocCount(nModules_);
+          for (auto& address : info->unmatched_) {
+            auto index = 0;
+            for (auto it = itBegin; it != itEnd; ++it, ++index) {
+              auto size = it->size();
+              if (0 != size) {
+                auto offset = it->findOffset(address);
+                if (offset != size) {
+                  moduleDeallocSize[index] += it->allocationSizes()[offset];
```

Comment:
How about something along
```suggestion
            auto found = std::find_if(itBegin, itEnd, [address&](auto const& elem){
              return elem.size() != 0 and elem.findOffset(address) != elem.size()
            });
            if (found != itEnd) {
              auto index = std::distance(itBegin, itEnd);
              moduleDeallocSize[index] += it->allocationSizes()[offset];
            }
```
?

Reply:
Your suggestion doesn't work since I need the return value from `findOffset`. Maybe with a mutable lambda which returns the value via a capture would work.

Reply:
Ah, good point. I'd guess that with ranges this could be expressed as a pipeline of transform and find_if, and maybe with the adaptors it wouldn't even lead to any additional calls to `findOffset()`. But I'm not sure if the exercise would really be worth it. Just a though.

Reply:
done